src/sinol_make/commands/run/__init__.py

```diff
@@ -495,11 +561,30 @@ def print_expected_scores(self, expected_scores):
     def validate_expected_scores(self, results):
         new_expected_scores = {} # Expected scores based on results
 
-        for solution in results.keys():
-            new_expected_scores[solution] = {
-                "expected": results[solution],
-                "points": self.calculate_points(results[solution])
-            }
+        def remove_points_from_results(results):
+            new_results = {}
+            for solution in results.keys():
+                new_results[solution] = {}
+                for group, item in results[solution].items():
+                    if isinstance(item, dict):
+                        new_results[solution][group] = item["status"]
+                    else:
+                        new_results[solution][group] = item
+            return new_results
+
+        if self.checker is None:
+            results = remove_points_from_results(results)
+            for solution in results.keys():
+                new_expected_scores[solution] = {
+                    "expected": results[solution],
+                    "points": self.calculate_points(results[solution])
+                }
+        else:
+            for solution in results.keys():
+                new_expected_scores[solution] = {
+                    "expected": remove_points_from_results(results[solution]),
+                    "points": self.calculate_points_checker(results[solution])
+                }
```

Comment:
I'm not sure, what the new code does and why is it needed to split the code into two cases?

Reply:
The `sinol_expected_scores` for packages with checkers look like this:
```yaml
  chk2.cpp:
    expected:
      1: {points: 25, status: OK}
      2: {points: 25, status: OK}
    points: 50
```

So now there is information about how much points solution scored on each test. Because of this the dictionaries look slightly different and the score calculation is different

Reply:
In the future we'll want to add the cutting of solution's scores when they are near the time limit, just like on OI contests in sio2. Such cases will also have status OK with partial points, but that won't be all cases. As such, it would be reasonable to print `{points: 25, status: OK}` only when the `points` are not equal to 100. This way of generating the output won't have to have different cases for the checkers.

Reply:
I changed how the points are calculated and saved now